File: 0922-sort-array-by-parity-ii/0922-sort-array-by-parity-ii.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParityII = function(nums) {
    
    let oddNums = [];
    let evenNums = [];

    evenNums = nums.filter((num) => num % 2 !== 1);
    oddNums = nums.filter((num) => num % 2 === 1)

    let evenIndex = 0;
    let oddIndex = 0;

    for(let i = 0 ; i < nums.length ; i = i + 2)
    {
        nums[i] = evenNums[evenIndex++];
        nums[i + 1] = oddNums[oddIndex++];
    }

    return nums;
};

File: 0922-sort-array-by-parity-ii/0922-sort-array-by-parity-ii.java
class Solution {
    public int[] sortArrayByParityII(int[] nums) {

        int length = nums.length;
        int evenNums[] = new int[length / 2];
        int oddNums[] = new int[length / 2];
        int evenIndex = 0;
        int oddIndex = 0;

        for(int num : nums)
        {
            if(num % 2 == 0 )
                evenNums[evenIndex++] = num;
            else
                oddNums[oddIndex++] = num;
        }

        evenIndex = 0;
        oddIndex = 0;

        for(int i = 0 ; i < length; i = i + 2)
        {
            nums[i] = evenNums[evenIndex++];
            nums[i + 1] = oddNums[oddIndex++];
        }

       return nums; 
    }   
}